esp-hal/src/rmt.rs

```diff
@@ -911,29 +949,38 @@ mod impl_for_chip {
                 peripheral,
                 channel0: ChannelCreator {
                     phantom: PhantomData,
+                    _guard: PeripheralGuard::new(crate::system::Peripheral::Rmt),
```

Comment:
I **think** the channelcreators don't need to enable the clocks, until the first channel is constructed.

Reply:
ah I remember - problem is a user can take the creator out, drop `Rmt` and then create a channel from the creator - we would then disable the already pre-configured peripheral

Reply:
But we probably don't need the guard on `Rmt` itself then